### PR TITLE
Fix: Fix the problems of the go linter 

### PR DESCRIPTION
**Pull request by nodauf**
**Description**

The last commit adds the Go linter to the CI. The goal of this PR is to fix all the problems with the Go linter to improve the code according to the Go standard. The following issues have been fixed:
- Typos
- Potential integer overflow
- Unused field in structure. The fields have been commented.
- Unchecked error

### CODE REVIEW

File: op-defender/psp_executor/api_test.go [hunk 1]
```diff
@@ -6,18 +6,19 @@ import (
...
+	"testing"
+
 	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
 	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/gorilla/mux"
-	"math/big"
```
Comment (nodauf) [inline op-defender/psp_executor/api_test.go:16]: The order of the imported package are important: https://golangci-lint.run/usage/linters/#gci

File: op-monitorism/liveness_expiration/monitor.go [hunk 2]
```diff
@@ -189,22 +191,24 @@ func (m *Monitor) Run(ctx context.Context) {
 		deadline_date := time.Unix(int64(deadline), 0)
 		formattedDate := deadline_date.Format("Monday, January 2, 2006")
 		// 4. Ensure that the invariant is not broken -> (block.timestamp + BUFFER > lastLive(owner) + livenessInterval) == true
-		if (deadline - now) < 0 {
```
Comment (nodauf) [inline op-monitorism/liveness_expiration/monitor.go:192]: This was a potential integer overflow as each variable was uint, `deadline - now` could never be less than 0. If now is greater than deadline, then the value will be close to `18446744073709551615`.
The correct way to do this operation is to use `bits.Sub64` and check the borrow value, if it is not 0 then something went wrong.

Comment (Ethnical) [reply op-monitorism/liveness_expiration/monitor.go:192]: Thanks greatly appreciated there!

File: op-defender/psp_executor/api_test.go (hunk 1 above)
Comment (Ethnical) [reply op-defender/psp_executor/api_test.go:16]: Nice catch!

**Review by Ethnical — APPROVED** [review]
Thanks this is really nice will merge it and see if the test are passing. 
I will just rename the variable `result` into the liveness_expiration to avoid confusion to `remainingTime`.

**Pull request merged**